File: Winter2021/lab-solutions/Lab7/RoomPersistenceLabSkeleton/app/src/main/java/mobiledev/unb/ca/roompersistencelab/dao/ItemDao.java
package mobiledev.unb.ca.roompersistencelab.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

import mobiledev.unb.ca.roompersistencelab.entity.Item;

/**
 * This DAO object validates the SQL at compile-time and associates it with a method
 */
@Dao
public interface ItemDao {
    // TODO Add app specific queries in here
    // Additional details can be found at https://developer.android.com/reference/android/arch/persistence/room/Dao
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insert(Item item);

    @Query("SELECT * FROM item_table WHERE name = :name ORDER BY num")
    List<Item> findItemsWithName(String name);
}

File: Fall2023/lab-solutions/Lab6/RoomPersistenceLabSkeleton/app/src/main/java/mobiledev/unb/ca/roompersistencelab/dao/ItemDao.kt
package mobiledev.unb.ca.roompersistencelab.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import mobiledev.unb.ca.roompersistencelab.entity.Item

/**
 * This DAO object validates the SQL at compile-time and associates it with a method
 */
@Dao
interface ItemDao {
    // TODO
    //  Add app specific queries for:
    //   1. Insert new record
    //   2. Query for items by name
    //  Additional details can be found at https://developer.android.com/reference/kotlin/androidx/room/Query
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insert(item: Item?)

    @Query("SELECT * FROM item_table WHERE name = :name ORDER BY num")
    fun findItemsWithName(name: String?): List<Item>
}
